### PR TITLE
[Cleanup] Remove Logo CTA Implementation On All Edit Pages With PDF Preview

### PR DESCRIPTION
**Pull request by Civolilah**
@beganovich @turbo124 The PR includes the implementation of the Remove Logo CTA for all edit pages with PDF preview, which encompasses `Invoices`, `Recurring Invoices`, `Quotes`, `Credits`, and `Purchase Orders`. Screenshot:

Screenshot 2023-12-13 at 17 57 00

@turbo124 Currently, this is shown only for `HOSTED` users with a `FREE PLAN`. Is it supposed to be like that?

Let me know your thoughts.

### CODE REVIEW

**Comment by turbo124** [comment]
@Civolilah in the case where this is an enterprise account, only the is_owner user should see this CTA.

**Comment by Civolilah** [comment]
> @Civolilah in the case where this is an enterprise account, only the is_owner user should see this CTA.

@turbo124 Ah, okay. I've made the change. If the user is `HOSTED` and has the `FREE PLAN`, and if they are the `OWNER` of the account, in that case, the CTA will be shown, otherwise, it will not. So, if someone adds a new user from the `enterprise` account to the `sub-account`, the user will NOT see the CTA. I did not have any enterprise account on the HOSTED platform to test this.

The condition is very straightforward, but if you have an enterprise account, please double-check that everything works perfectly fine.

File: src/components/RemoveLogoCTA.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { useCurrentUser } from '$app/common/hooks/useCurrentUser';
+import { useAdmin } from '$app/common/hooks/permissions/useHasPermission';
+
+export function RemoveLogoCTA() {
+  const [t] = useTranslation();
+  const user = useCurrentUser();
+
+  const { isOwner } = useAdmin();
+
+  return (
```
Comment (beganovich) [inline src/components/RemoveLogoCTA.tsx:26]: You can get way cleaner implementation by early return and escape ? and {} hell.

```tsx
if (!proPlan() && !enterprisePlan() && isOwner) {
...
}

return null;

File: src/pages/credits/edit/Edit.tsx [hunk 2]
```diff
@@ -181,6 +182,8 @@ export default function Edit() {
               endpoint="/api/v1/live_preview?entity=:entity"
             />
           )}
+
+          <RemoveLogoCTA />
```
Comment (beganovich) [inline src/pages/credits/edit/Edit.tsx:186]: Is there a reason we didn't bake this directly into the InvoicePreview component, or the component one-step lower?

**Review by beganovich — CHANGES_REQUESTED** [review]

File: src/components/RemoveLogoCTA.tsx (hunk 1 above)
Comment (Civolilah) [reply src/components/RemoveLogoCTA.tsx]: @beganovich Sure thing! I refactored it, it looks cleaner now!

**Review by beganovich — APPROVED** [review]

**Pull request merged**